src/taskUpdate.js

```diff
@@ -0,0 +1,105 @@
+  }
+
+  // Function displayTasks.
+  displayList = () => {
+    const newInput = document.querySelector('#new-input');
+    newInput.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter' && newInput.value) {
+        this.addTask(newInput.value);
+        newInput.value = '';
+      }
+    });
+
+    // Add New Task In List.
+    const toDoContainer = document.querySelector('#todo-list');
+    toDoContainer.innerHTML = '';
+    this.toDoList.forEach((item) => {
+      const listItem = document.createElement('li');
+      if (item.completed === true) {
+        listItem.classList.add('checked');
+      }
+      listItem.innerHTML = `
+        <div class="label-container">
+          <label for="checkbox">
+            <input type="checkbox" id="${item.index}" ${item.completed ? 'checked' : ''}>
+          </label>
+          <label for="to-do">
+            <input class="to-do" type="text" value="${item.description}">
+          </label>
+        </div>
+        <i class="fas fa-trash-can delete-task"></i>`;
+      toDoContainer.appendChild(listItem);
+    });
+
+    // Update Task.
+    const updateTask = document.querySelectorAll('.to-do');
+    updateTask.forEach((input, index) => {
+      input.addEventListener('keypress', (ToDo) => {
+        if (ToDo.key === 'Enter' && input.value) {
+          this.updateTask(input.value, index);
+        }
+      });
+      input.addEventListener('change', () => {
+        if (input.value) {
+          this.updateTask(input.value, index);
+        }
+      });
+    });
+
+    // Delete Task.
+    const deleteBtn = document.querySelectorAll('.delete-task');
+    deleteBtn.forEach((button, index) => {
+      button.addEventListener('click', () => {
+        this.removeTask(index);
+      });
+    });
+
+    // Mark A Task As Completed
+    userAction.completeTask(this.toDoList);
+
+    // Clear All Completed Tasks
+    userAction.clearAllCompleted(this.toDoList);
+
+    // Refresh The To-Do List.
+    const refreshBtn = document.querySelector('#refresh');
+    refreshBtn.addEventListener('click', () => {
+      document.location.reload();
+    });
+  }
+
+  // Add Task In Local Storage.
+  addTask = (value) => {
+    const newTask = {
+      description: value,
+      completed: false,
+      index: this.toDoList.length,
+    };
+    this.toDoList.push(newTask);
+    localStorage.setItem('tasks', JSON.stringify(this.toDoList));
+    this.displayList();
+  }
+
+  // Update Task In Local Storage.
+  updateTask = (value, index) => {
+    this.toDoList[index].description = value;
+    localStorage.setItem('tasks', JSON.stringify(this.toDoList));
+    this.displayList();
+  }
+
+  // Remove Task from Local Storage.
+  removeTask = (index) => {
+    this.toDoList.splice(index, 1);
+    for (let i = 0; i < this.toDoList.length; i += 1) {
+      this.toDoList[i].index = i;
+    }
+    localStorage.setItem('tasks', JSON.stringify(this.toDoList));
+    this.displayList();
+  }
+}
```

Comment:
- [ ] It appears that the code you provided includes three functions for adding, updating, and deleting tasks from local storage. These functions are similar in that they all involve updating the toDoList array and saving the updated array to local storage using the localStorage.setItem method.

- [ ]  The DRY principle suggests that code should be written in a way that avoids repeating the same information or functionality in multiple places.

- [ ] In this case, it might be possible to refactor the code to make it more DRY by extracting the code for saving the toDoList array to local storage into a separate function that can be called from all three of these functions. This would allow you to avoid repeating the same code for saving the array to local storage in multiple places.

Reply:
Thanks for your review. I've applied the DRY principle by creating a separate function to update the toDoList `Array [...]` as per your suggestion.